### ISSUE TITLE
[indent] <Indentation rule conflict with prettier (type declaration)>

### ISSUE THREAD

**Issue by grin0c** [typescript-eslint/typescript-eslint#1161]
**Repro**

```JSON
{
  "rules": {
    "max-len": ["warn", { "code": 100 }],
    "@typescript-eslint/indent": ["error", 2]
  }
}
```

```TS
export const LetterOutgoingColumns: ColumnsOptions<
  Pick<LetterEntity, LetterOutgoingColumnsKeys>
> = {
  recipient: {
    type: 'string',
    title: 'Address',
    validation: {
      schema: Joi.string()
        .length(256)
        .optional(),
      level: ValidationLevel.Wrong,
    },
  },
}
```

**Expected Result**

**Actual Result**
When prettier is followed:
```
error  Expected indentation of 0 spaces but found 2  @typescript-eslint/indent
```

When eslint is followed:
```
warning  Insert `··`  prettier/prettier
```

**Additional Info**
<img width="667" alt="Снимок экрана 2019-10-30 в 10 54 08" src="https://user-images.githubusercontent.com/7963898/67839059-c29c2f00-fb03-11e9-8f3e-1c6059f7b27e.png">
<img width="553" alt="Снимок экрана 2019-10-30 в 10 54 39" src="https://user-images.githubusercontent.com/7963898/67839062-c5971f80-fb03-11e9-8bfd-f1768de2227e.png">

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `^2.6.0` |
| `@typescript-eslint/parser`        | `^2.6.0` |
| `TypeScript`                       | `^3.6.4` |
| `ESLint`                           | `^6.6.0` |
| `node`                             | `10.15.0` |
| `npm`                              | `6.4.1` |

**Comment by bradzacher** [comment]
There is no expected compatibility between prettier and eslint styling rules.
The two do not use the same algorithms, and have no intention to - prettier is intentionally opinionated and has few options, lint rules are unopinionated with many options.

You should use one or the other.
Prettier provides configs to help you turn off all stylistic eslint rules that might conflict with it.

**Comment by bradzacher** [comment]
https://github.com/prettier/eslint-config-prettier#readme
